python/ray/serve/_private/application_state.py

```diff
@@ -209,6 +213,23 @@ def get_application_status_info(self) -> ApplicationStatusInfo:
             deployment_timestamp=self.deployment_timestamp,
         )
 
+    def list_deployment_details(self) -> Dict[str, DeploymentDetails]:
```

Comment:
should we move the function logics into the deployment_state_manager()?
In application_state.py
```
def list_deployment_details(self):
   return {name : self.deployment_state_manager.get_deployment_details(deployment_name) for name in self.get_all_deployments()}
```

in the deployment_state.py
```
def get_deployment_details(name: str):
   <logics you have here>
```

Application doesn't need to care about how to construct the DeploymentDetails. In the future, we can change internal logics in deployment_state easily without changing application_state code.

Reply:
That's a good suggestion, thanks! I've changed it.